### PR TITLE
rework rccl BUILD rules to match more closely with upstream

### PR DESCRIPTION
**Pull request by jeffdaily**
- remove `nccl_config`
- use `if_cuda_or_rocm` in place of `if_cuda`, not `if_nccl`
- use `if_cuda`/`if_rocm`, not if_cuda/rocm_is_configured
- misc changes to match upstream files

### CODE REVIEW

File: tensorflow/core/kernels/BUILD [hunk 1]
```diff
@@ -38,8 +37,11 @@ load(
     "if_mkl_ml",
     "mkl_deps",
 )
-load("@local_config_cuda//cuda:build_defs.bzl", "if_cuda", "if_cuda_is_configured")
-load("@local_config_rocm//rocm:build_defs.bzl", "if_rocm", "if_rocm_is_configured")
+load("@local_config_cuda//cuda:build_defs.bzl", "if_cuda")
+load(
```
Comment (whchung) [inline tensorflow/core/kernels/BUILD]: @jeffdaily could you run this file with `buildifier -v -mode=check`? Please refer to `tensorflow/tools/ci_build/ci_sanity.sh`.

Comment (jeffdaily) [reply tensorflow/core/kernels/BUILD]: I ran `buildifier -v -mode=diff ./tensorflow/core/kernels/BUILD` but none of the lines I changed were indicated by the diff.

Comment (jeffdaily) [reply tensorflow/core/kernels/BUILD]: Pardon, it does add one empty line to the file between the two virtual targets I added.  How would you like me to proceed?  Would you like me to run `buildifier` against all BUILD files I changed?

Comment (whchung) [reply tensorflow/core/kernels/BUILD]: I was wondering would `buildifier` help re-indent these `load`s. seems it doesn't. could you re-indent them?

File: tensorflow/core/kernels/BUILD [hunk 2]
```diff
@@ -193,6 +195,16 @@ tf_cc_test(
     ],
 )
 
+# virtual targets since nested select statements not possible
```
Comment (whchung) [inline tensorflow/core/kernels/BUILD]: nice bazel usage here!

Comment (jeffdaily) [reply tensorflow/core/kernels/BUILD]: Thanks!  I can't take full credit.  I got the idea from https://github.com/bazelbuild/bazel/issues/1623#issuecomment-369909088

**Comment by jeffdaily** [comment]
@whchung  If the previous CI run completes, I would be okay with not needed the next commit's CI run since it was a whitespace change and I verified it locally.  I'll leave it up to you.  Depends how busy our CI is I suppose.

**Pull request merged**